charts/cryostat/tests/deployment_test.yaml

```diff
@@ -0,0 +1,164 @@
+          path: spec.template.spec.containers[?(@.name=='cryostat')].env[?(@.name=='STORAGE_BUCKETS_ARCHIVES_NAME')].value
+          value: "archivedrecordings"
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].livenessProbe.httpGet.path
+          value: "/health/liveness"
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].startupProbe.httpGet.path
+          value: "/health/liveness"
+
+  - it: should validate volumes and volume mounts
+    asserts:
+      - exists:
+          path: spec.template.spec.volumes[?(@.name=='alpha-config')]
+      - exists:
+          path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].volumeMounts[?(@.mountPath=='/etc/oauth2_proxy/alpha_config')]
+
+  - it: should validate container ports and network settings
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat')].ports[0].containerPort
+          value: 8181
+      - equal:
+          path: spec.template.spec.containers[?(@.name=='cryostat-storage')].ports[0].containerPort
+          value: 8333
+
+  - it: should set image pull secrets if specified
+    set:
+      imagePullSecrets:
+        - name: myregistrykey
+    asserts:
+      - equal:
+          path: spec.template.spec.imagePullSecrets[0].name
+          value: "myregistrykey"
+
+  - it: should configure the OpenShift OAuth Proxy when authentication.openshift is enabled
+    set:
+      authentication.openshift.enabled: true
+    asserts:
+      - matchRegex:
+          path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].image
+          pattern: "quay.io/cryostat/openshift-oauth-proxy:.*"
+
+  - it: should configure the OAuth2 Proxy when OpenShift OAuth is disabled and OAuth2 is enabled
+    set:
+      authentication.openshift.enabled: false
+      authentication.oauth2.enabled: true
+    asserts:
+    - exists:
+        path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')]
+    - matchRegex:
+        path: spec.template.spec.containers[?(@.name=='cryostat-authproxy')].image
+        pattern: "quay.io/oauth2-proxy/oauth2-proxy:.*"
+
+  - it: should apply Kubernetes specific settings when configured
+    set:
+      nodeSelector:
+        disktype: ssd
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: disktype
+                operator: In
+                values:
+                - ssd
+      tolerations:
+      - key: "key"
+        operator: "Equal"
+        value: "value"
+        effect: "NoSchedule"
+    asserts:
+      - exists:
+          path: spec.template.spec.nodeSelector
+      - exists:
+          path: spec.template.spec.affinity
+      - exists:
+          path: spec.template.spec.tolerations
+
+  - it: should ensure the datasource container has correct environment variables and resources
```

Comment:
If we are setting up the tests based on setting types (i.e. security context, env vars, pull policies), then each test should validate all containers. So this can be merged with above test case: `should validate all container settings including environment variables...`